### PR TITLE
UPERF: Fix node name truncation to avoid ending with a dot  '.' char

### PR DESCRIPTION
**Pull request by HughNhan**
### Description
Internally UPERF benchmark task(s) may truncate a node name and combine it with a suffix to create other name(s) for internal use.  The truncation is based on a fixed length and hence could result in the truncated portion ended with a dot '.' character. Subsequently the combined name may be an illegal name e.g. "abc.-0-suffix" per RFC 1123
### Fixes
During truncation if the corner case happens, truncate one more char to trim off the dot.

### Test
The diff works on local unit-test and on the original system reported  by issue #644

### CODE REVIEW

**Comment by sjug** [comment]
Thanks for the PR @HughNhan , can you please standardize the spacing `worker_node_list[ node_idx_item ]`?

**Comment by HughNhan** [comment]
> Thanks for the PR @HughNhan , can you please standardize the spacing `worker_node_list[ node_idx_item ]`?

Can you elaborate?

File: roles/uperf/templates/server.yml.j2 [hunk 1]
```diff
@@ -7,7 +7,11 @@ items:
   - kind: Job
     apiVersion: batch/v1
     metadata:    
-      name: 'uperf-server-{{worker_node_list[ node_idx_item ] | truncate(27,true,'') }}-{{ item }}-{{ trunc_uuid }}'
+{% if (worker_node_list[ node_idx_item]|length>27) and (worker_node_list[ node_idx_item ][26] == '.') %}
```
[the remark this answers is not in the document]
Comment (HughNhan) [reply roles/uperf/templates/server.yml.j2:10]: We are about to truncate the name to 27 char in length (and then append some suffix) . If the 27th char is a dot, truncate it to 26 char length, one pass the dot. If the original name is 27 char in length or less, no need to truncate.

**Review by rsevilla87 — APPROVED** [review]
lgtm

**Comment by sjug** [comment]
> Can you elaborate?

roles/uperf/templates/server.yml.j2:
```
10: {% if (worker_node_list[ node_idx_item]|length>27) and (worker_node_list[ node_idx_item ][26] == '.') %}
25: {% if (worker_node_list[ node_idx_item]|length>27) and (worker_node_list[ node_idx_item ][26] == '.') %}
```

roles/uperf/templates/service.yml.j2:
15: {% if (worker_node_list[ node_idx_item]|length>27) and (worker_node_list[ node_idx_item ][26] == '.') %}
26: {% if (worker_node_list[ node_idx_item]|length>27) and (worker_node_list[ node_idx_item ][26] == '.') %}

The first condition is `[ node_idx_item]` (no space after the variable name), the second condition is `[ node_idx_item ]` (correct).

**Review by sjug — APPROVED** [review]
/lgtm
